redact/tools/redact_folder.py

```diff
@@ -94,22 +89,25 @@ def redact_folder(
     )
 
     log.info(f"Starting {n_parallel_jobs} parallel jobs to anonymize files ...")
-    _parallel_map(
+    return _parallel_map(
         func=worker_function, items=relative_file_paths, n_parallel_jobs=n_parallel_jobs
     )
 
 
-def _parallel_map(func, items: List, n_parallel_jobs=1):
-    if n_parallel_jobs <= 1:
-        # Anonymize one item at a time. In principle, the ThreadPoolExecutor could do this with one worker only. But
-        # this ways we don't risk losing exceptions in the thread.
-        list(tqdm.tqdm(map(func, items), total=len(items)))
-    else:
-        # Anonymize files concurrently
-        with concurrent.futures.ThreadPoolExecutor(
-            max_workers=n_parallel_jobs
-        ) as executor:
-            list(tqdm.tqdm(executor.map(func, items), total=len(items)))
+def _parallel_map(func, items: List, n_parallel_jobs=1) -> Tuple[Dict[str, Any], ...]:
+    results = {}
+    exceptions = {}
+
+    with ThreadPoolExecutor(max_workers=n_parallel_jobs) as executor:
```

Comment:
Do I understand correctly that we will only log redact errors once all files in the folder are processed?

Reply:
yes, I did not want to print excessive information and Interrupt other files processing, 
that is why, it will just print all issues once finished